app/assets/v2/js/grants/cart/polkadot_extension.js

```diff
@@ -0,0 +1,145 @@
+const initPolkadotConnection = async(grant, vm) => {
+
+  // step 1: check if web3 is injected
+  if (!polkadot_utils.isWeb3Injected) {
+    vm.updatePaymentStatus(grant.grant_id, 'failed');
+    _alert({ message: `Please ensure your Polkadot One wallet is installed and unlocked`}, 'error');
+    return;
+  }
+
+  // step 2: init connection based on token
+  let polkadot_endpoint;
+  let decimals;
+  let format;
+  if (grant.grant_donation_currency == 'KSM') {
+    polkadot_endpoint = KUSAMA_ENDPOINT;
+    decimals = 12;
+    format = 0;
+  } else if (grant.grant_donation_currency == 'DOT') {
+    polkadot_endpoint = POLKADOT_ENDPOINT;
+    decimals = 10;
+    format = 2;
+  }
+
+  polkadot_utils.connect(polkadot_endpoint).then(res =>{
+    polkadot_extension_dapp.web3Enable('gitcoin').then(() => {
+      initComplete(null, grant, vm);
+    }).catch(err => {
+      initComplete(err);
+    });
+  });
+
+  // step 3: allow user to select address on successful connection
+  async function initComplete(error, grant, vm) {
+
+    if (error) {
+      vm.updatePaymentStatus(grant.grant_id, 'failed');
+      _alert('Please ensure you\'ve connected your polkadot extension to Gitcoin', 'error');
+      console.log(error);
```

Comment:
that was intentional ! just so that the error stays ! cause this is behind a staff flag -> will remove this next week once folks test and we are sure it's smooth